app/Http/Controllers/Api/VerificationController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Hash;
+use Libraries\Otp;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Crypt;
+
+class VerificationController extends Controller {
+
+	private const EXPIRATION = 1800; // 30 minutes
+
+	public function store() {
+		$key = Crypt::decrypt(request('token'));
+		$otp = new Otp(Cache::get($key));
+
+		request()->validate( [
+			'token' => 'required',
+			'code'  => [
+				'required',
+				function ( $attribute, $value, $fail ) use ($otp) {
+					if ( ! $otp->isValid( $value ) ) {
+						$fail( "Verification code was expired or invalid. Please login or resend verification code" );
+					}
+				}
+			]
+		]);
+
+		$token = Hash::make($key);
+		Cache::put($token, explode('-', $key)[1], self::EXPIRATION);
```

Comment:
This extracts the mobile number from the previous key, may I suggest to declare a variable here to make it more readable